pkg/mtd/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mtd
+
+// Info provides information about an active MTD.
+type Info interface {
+	DevName() string
+}
+
+// Flasher defines the interface to flash drivers.
+//
+// Many devices must have lazy writes; SyncWrite should always be called
+// after a sequence of QueueWrite commands.  Close should return an
+// error if there are queued write commands. To erase a device, one
+// calls chip Blank(), QueueWrite(), and SyncWrite(). The operators are deined
+// for the Flasher, not the Chipper, since flashing can involve
+// driver-level operations such as unlocking protection bits on
+// a bridge that are more than just a chip operation.
+type Flasher interface {
+	ReadAt([]byte, int64) (int, error)
+	QueueWrite([]byte, int64) (int, error)
+	SyncWrite() error
+	Close() error
+}
+
+// Chipper is an interface to an individual chip.
+//
+// Blank will return a []byte full of "zero" values optimized for a
```

Comment:
move these individual comments to the funcs, i.e.

```go
// Chipper is a ...
type Chipper interface {
  // Blank returns a []byte full of "zero" values ...
  Blank() []byte

...
}
```

Reply:
done